### PR TITLE
SqueezeNet 1.0 and 1.1 models

### PR DESCRIPTION
**Pull request by Maratyszcza**
SqueezeNet 1.0 and 1.1 models. I'm still working on getting them converge on ImageNet, but as several people contacted me about SqueezeNet for PyTorch, I think its useful to share it publicly.

### CODE REVIEW

**Comment by Maratyszcza** [comment]
I successfully trained SqueezeNet 1.0 on ImageNet with the following parameters: `--arch squeezenet1_0 --epochs 90 --lr 0.01 --weight-decay 0.0002 --batch-size 128`.
Accuracy: `Prec@1 57.868 Prec@5 80.424`

**Comment by apaszke** [comment]
Awesome! Is the PR ready?

**Comment by Maratyszcza** [comment]
There is SqueezeNet 1.1 not train as well

**Comment by Maratyszcza** [comment]
Finished training both SqueezeNet 1.0 and 1.1. Trained for 100 epochs with `--epochs 100 --lr 0.01 --weight-decay 0.0002 --batch-size 128`

| Model          | Top-1 accuracy | Top-5 accuracy |
| -------------- |:--------------:|:--------------:|
| SqueezeNet 1.0 | 58.000%        | 80.488%        |
| SqueezeNet 1.1 | 58.184%        | 80.514%        |

After @soumith uploads the models to AWS, I'll add `model_zoo` links, and we'd be ready to merge.

**Comment by apaszke** [comment]
Great! If you put them somewhere or send them to me, I can upload them to AWS too.

**Comment by soumith** [comment]
finished uploading them:

```
https://s3.amazonaws.com/pytorch/models/squeezenet1_0-0ef9bd12.pth.tar
https://s3.amazonaws.com/pytorch/models/squeezenet1_1-f0e1d24d.pth.tar
```

**Comment by Maratyszcza** [comment]
@apaszke @soumith Ready to merge!

File: torchvision/models/squeezenet.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+                Fire(256, 32, 128, 128),
+                nn.MaxPool2d(kernel_size=3, stride=2, ceil_mode=True),
+                Fire(256, 48, 192, 192),
+                Fire(384, 48, 192, 192),
+                Fire(384, 64, 256, 256),
+                Fire(512, 64, 256, 256),
+            )
+        self.classifier = nn.Sequential(
+            nn.Dropout(p=0.5),
+            self.final_conv,
```
Comment (apaszke) [inline torchvision/models/squeezenet.py]: Can you create the module in here, instead of putting it both in the main network, and in that Sequential? We don't support reusing modules in different parts of the network (you can just call the same one multiple times).

File: torchvision/models/squeezenet.py [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+            nn.ReLU(inplace=True),
+            nn.AvgPool2d(13)
+        )
+
+        import math
```
Comment (apaszke) [inline torchvision/models/squeezenet.py]: Can you move that import to the top of the script?

File: torchvision/models/squeezenet.py (hunk 1 above)
Comment (Maratyszcza) [inline torchvision/models/squeezenet.py]: If I remove `final_conv` now, the pre-trained models wouldn't load

File: torchvision/models/squeezenet.py (hunk 2 above)
Comment (Maratyszcza) [inline torchvision/models/squeezenet.py]: Ok, sure

File: torchvision/models/squeezenet.py (hunk 1 above)
Comment (apaszke) [inline torchvision/models/squeezenet.py]: True. It should be enough to remove the keys of `final_conv`, as they're duplicated. I can do that tomorrow.
